client/driver/executor/executor.go

```diff
@@ -0,0 +1,217 @@
+package executor
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"syscall"
+	"time"
+
+	cgroupConfig "github.com/opencontainers/runc/libcontainer/configs"
+
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/driver/env"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// ExecutorContext is a wrapper to hold context to configure the command user
+// wants to run
+type ExecutorContext struct {
+	TaskEnv          *env.TaskEnvironment
+	AllocDir         *allocdir.AllocDir
+	TaskName         string
+	TaskResources    *structs.Resources
+	FSIsolation      bool
```

Comment:
@dadgar ResourceIsolation doesn't convey the meaning well. Usually, something like cgroup is known as "limit" and when a process can't see the whole environment of it's host it's called isolation( anything related to name space)